lte/gateway/c/core/oai/test/amf/test_amf.cpp

```diff
@@ -402,26 +402,27 @@ TEST(test_amf_nas5g_pkt_process, test_amf_service_accept) {
           .pdu_re_activation_status.pduSessionReActivationResult,
       PDU_SESSION_ID);
 }
-TEST(test_amf_data_struct, test_ue_context_creation) {
-  ue_m5gmm_context_s* ue_context = nullptr;
 
-  ue_context = amf_create_new_ue_context();
+class AmfUeContextTest : public ::testing::Test {
+ protected:
+  ue_m5gmm_context_s* ue_context;
+
+  virtual void SetUp() { ue_context = amf_create_new_ue_context(); }
+  virtual void TearDown() { delete ue_context; }
+};
+
+TEST_F(AmfUeContextTest, test_ue_context_creation) {
   EXPECT_TRUE(nullptr != ue_context);
   EXPECT_TRUE(0 == ue_context->amf_teid_n11);
```

Comment:
why did you move the tested function call to SetUp() ? This does not make much sense to me.

Reply:
@ulaskozat,
we would like to introduce text fixtures to move reusable code.
and existing multiple test cases having ue_context allocation and deallocation.

Please provide your feedback if the improvement is needed here.

Reply:
Hmm. It is a bit odd that `test_ue_context_creation` is not actually calling any function. Is there a plan to add more test cases here?

Reply:
Okay let's leave it as is for now and we can reassess later based on the new test cases being added.